src/main/scala/org/broadinstitute/dsde/firecloud/model/Workspace.scala

```diff
@@ -1,108 +1,39 @@
-
-case class WorkspaceName(
-  namespace: Option[String] = None,
-  name: Option[String] = None)
-
-case class WorkspaceEntity(
-  namespace: Option[String] = None,
-  name: Option[String] = None,
-  createdDate: Option[String] = None,
-  createdBy: Option[String] = None,
-  attributes: Option[AttributeMap] = None)
+import org.broadinstitute.dsde.rawls.model.Attributable.AttributeMap
+import org.broadinstitute.dsde.rawls.model._
+import org.joda.time.DateTime
 
 case class WorkspaceCreate(
   namespace: String,
   name: String,
   attributes: AttributeMap,
   isProtected: Option[Boolean] = Some(false))
 
-case class RawlsWorkspaceCreate(
-  namespace: String,
-  name: String,
-  attributes: AttributeMap,
-  realm: Option[Map[String, String]] = None) {
-  def this(wc: WorkspaceCreate) =
-    this(wc.namespace, wc.name, wc.attributes,
-      if (wc.isProtected.getOrElse(false))
-        Some(Map("realmName" -> FireCloudConfig.Nih.rawlsGroupName))
-      else None)
-}
-
-case class RawlsWorkspaceResponse(
-  accessLevel: String,
-  canShare: Option[Boolean] = None,
-  workspace: RawlsWorkspace,
-  workspaceSubmissionStats: SubmissionStats,
-  owners: List[String])
-
-case class RawlsWorkspace(
-  workspaceId: String,
-  namespace: String,
-  name: String,
-  isLocked: Option[Boolean] = None,
-  createdBy: String,
-  createdDate: String,
-  lastModified: Option[String] = None,
-  attributes: AttributeMap,
-  bucketName: String,
-  accessLevels: Map[String, Map[String, String]],
-  realm: Option[Map[String, String]])
-
-case class RawlsEntity(name: String, entityType: String, attributes: AttributeMap)
-
-case class SubmissionStats(
-  lastSuccessDate: Option[String] = None,
-  lastFailureDate: Option[String] = None,
-  runningSubmissionsCount: Int)
-
-case class UIWorkspaceResponse(
-  accessLevel: Option[String] = None,
-  canShare: Option[Boolean] = None,
-  workspace: Option[UIWorkspace] = None,
-  workspaceSubmissionStats: Option[SubmissionStats] = None,
-  owners: Option[List[String]] = None) {
-  def this(rwr: RawlsWorkspaceResponse) =
-    this(Option(rwr.accessLevel), rwr.canShare, Option(new UIWorkspace(rwr.workspace)), Option(rwr.workspaceSubmissionStats), Option(rwr.owners))
-}
-
-case class UIWorkspace(
-  workspaceId: String,
-  namespace: String,
-  name: String,
-  isLocked: Option[Boolean] = None,
-  createdBy: String,
-  createdDate: String,
-  lastModified: Option[String] = None,
-  attributes: AttributeMap,
-  bucketName: String,
-  accessLevels: Map[String, Map[String, String]],
-  realm: Option[Map[String, String]],
-  isProtected: Boolean) {
-  def this(rw: RawlsWorkspace) =
-    this(rw.workspaceId, rw.namespace, rw.name, rw.isLocked, rw.createdBy, rw.createdDate,
-      rw.lastModified, rw.attributes, rw.bucketName, rw.accessLevels, rw.realm,
-      rw.realm.flatMap(_.get("realmName").map(_ == FireCloudConfig.Nih.rawlsGroupName)).getOrElse(false))
+object WorkspaceCreate {
```

Comment:
This generally looks better. (Can this just be called `Workspace` now?) I do worry a little about the realm parts... not because I see a problem with these changes but because I know how fragile this code was previously.

@MatthewBemis, can you have a look specifically at the changes here around determining `isProtected`? In light of this, I'm very glad for the approach that we took with testing today!